Add configuration guide

xtuc opened the pull request.

hzoo commented:
Deploy preview for *babel-new* ready!

Built with commit 91fb1fb9eca92e2deaab18d1c70146b7e489f524

https://deploy-preview-1679--babel-new.netlify.com

xtuc commented on docs/babelconfigjs.md:
```diff
@@ -0,0 +1,97 @@
+
+## Lookup behavior
+
+The lookup behavior is simple, Babel will only look for a `babel.config.js` in the root directory (where your package.json is by default). You can configure if needed, see [Specifying the root folder](babelconfigjs.md#specifying-the-root-folder).
```
Remove simple

alexzherdev commented on docs/babelconfigjs.md:
```diff
@@ -0,0 +1,97 @@
+
+### Environment
+
+A conditional configuration is useful when you want to use certain plugins/presetns based on the current environement, for example:
```
typos: `presets`, `environment`

alexzherdev commented on docs/babelconfigjs.md:
```diff
@@ -0,0 +1,97 @@
+
+### `api.env()`
+
+Returns Babel's environement, you can configure it using the `BABEL_ENV` environement variable.
```
2x`environment`

xtuc replied:
Do you have a suggestion?

alexzherdev replied:
Oh, I meant there's 2 typos in `environment` here

xtuc replied:
Yes but I couldn't find a better wording for that, do you have a suggestion?

alexzherdev replied:
This actually sounds clear to me even with the repetitiveness.

alexzherdev commented on docs/babelconfigjs.md:
```diff
@@ -0,0 +1,97 @@
+
+## Extending other `.babelrc`
+
+To allow people to opt into `.babelrc` usage, potentially for local development; you can specifiy a list of `.babelrc` to use.
```
`specify`

xtuc commented:
Thanks @alexzherdev

hzoo commented on docs/babelconfigjs.md:
```diff
@@ -0,0 +1,97 @@
+id: babelconfigjs
+---
+
+## Lookup behavior
```
Should we explain the file or use if they are linked directly to this page?

hzoo commented on docs/babelrc.md:
```diff
@@ -3,67 +3,17 @@ title: .babelrc
+## Lookup behavior
 
-## Use via `@babel/cli`
+Babel will look for a `.babelrc` in the current directory (which defaults to cwd).
```
For lookup I thought you would want to explain how it keeps looking up the folder until it hits package.json?

xtuc replied:
According to the pr (https://github.com/babel/babel/pull/7358) it doesn't do that anymore? See "When searching for .babelrc files for a given file, stop at package.json"

hzoo replied:
Oh Yeah that's what I meant, about it stopping at package.json?

hzoo commented on docs/babelconfigjs.md:
```diff
@@ -0,0 +1,97 @@
+
+## Lookup behavior
+
+The lookup behavior is direct as Babel will only look for a `babel.config.js` in the root directory (where your `package.json` is by default). If configuration is needed, see [specifying the root folder](babelconfigjs.md#specifying-the-root-folder).
```
Maybe below we can create a file diagram to explain how it ignores the other config files

```
a/
--babel.config.js
--b/
----.babelrc // ignored
----b.js // current file
```

something like that kind of example

hzoo commented on docs/babelconfigjs.md:
```diff
@@ -0,0 +1,97 @@
+
+### Environment
+
+A conditional configuration is useful when you want to use certain plugins/presets based on the current environment, for example:
```
should mention that `"env"` still works fine for people, not sure if it should go here or elsewhere

xtuc replied:
Yes, I wasn't sure if we wanted to include it here. With this format I'd like to promote that you can use `process.env` directly instead, which means that the merging strategy is done on their side. I think it's better to avoid mentioning it. Thoughts?

alexzherdev commented on docs/babelconfigjs.md:
```diff
@@ -0,0 +1,97 @@
+
+## Specifying the root folder
+
+You can pass a `root` in the Babel configuation, see [@Babel/core's documentation](babel-core.md#options).
```
`configuration`

hzoo commented on docs/babelconfigjs.md:
```diff
@@ -0,0 +1,97 @@
+module.exports = function () {
+  const plugins = [ ... ];
+
+  if (true === false) {
```
We should use something real here? Or just combine it with the environment example

hzoo commented on docs/configuration.md:
```diff
@@ -0,0 +1,111 @@
+
+> [`.babelrc`](#babelrc) is for you!
+
+- The guy fieri is your hero?
```
haha not sure this really fits but I understand the sentiment

xtuc commented:
@hzoo I added a diagram about the merging behavior for `babel.config.js`, one for new resolution of `.babelrc` and changed the dynamic configuration example (access to a local file)

hzoo commented:
Cool I'll merge and we can iterate
